src/Sulu/Bundle/RouteBundle/Tests/Unit/Routing/RouteProviderTest.php

```diff
@@ -354,4 +354,33 @@ public function testGetRouteCollectionForRequestEndingDot()
 
         $this->assertCount(0, $collection);
     }
+
+    public function testGetRouteCollectionForRequestWithoutFormatExtension()
+    {
+        $request = $this->prophesize(Request::class);
+        $request->getPathInfo()->willReturn('/de/test');
```

Comment:
i think the reuqest will return `/de/test.json` in a real-world example? didn't it?

Reply:
The use case is actually requesting a JSON page without using the `.json` extension. 
The requested url for example is:
```
https://www.example.com/en/articles/foo-bar
```
And in my own request analyser the request format is set to json:
```
 $request->setRequestFormat('json');
```
So the path info should not return a path with a `.json` extension in this case? What you describe is already [covered in another test](https://github.com/jaapromijn/sulu/blob/146c7d27b5ddb9923088cbfda7a248d0a5fc2de2/src/Sulu/Bundle/RouteBundle/Tests/Unit/Routing/RouteProviderTest.php#L153), I think.

Reply:
correct :) haven't recognized this - this PR is good! thanks for contribution

Reply:
👍